### PR TITLE
Build detection content for Defense Evasion : Modify Authentication P…

### PR DESCRIPTION
**Pull request by Avni3**

### CODE REVIEW

File: workbooks/Advanced Threat Detection/IAM Monitoring/mfa_disabled_for_gcp_user.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+- Disable the compromised account or revoke the user's access and notify other associated accounts/users
+
+- Implement additional security measures by resetting the password for other associated accounts and implementing stricter access controls
+
+### Post Incident handling documentation
+
+Note:- Update according to the Incidents handled/SOP"
+
+  raw_event_query: SELECT * FROM IAM WHERE $Action="UNENROLL_USER_FROM_STRONG_AUTH"
```
Comment (aman1808) [inline workbooks/Advanced Threat Detection/IAM Monitoring/mfa_disabled_for_gcp_user.yml:49]: update sourcename in the raw_query as well and use new dql

File: workbooks/Advanced Threat Detection/IAM Monitoring/mfa_disabled_for_gcp_user.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+entity_type: workbooks
+blocks:
+- type: dql
+  query: stream=iam where action='UNENROLL_USER_FROM_STRONG_AUTH' | groupby user,
+    targetuser, status | limit 10
+- type: signal
+  DetectionName: MFA Disabled for GCP User
+  DetectionTactic: Defense Evasion
+  DetectionTechnique: Modify Authentication Process
+  DetectionScore: '5'
```
Comment (aman1808) [inline workbooks/Advanced Threat Detection/IAM Monitoring/mfa_disabled_for_gcp_user.yml:16]: revisit the numbers on score, severity and confidence as this use case looks more serious

File: workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+id: dcc3e44bd75147339d0a339971f25e72
+name: Successfull Login from Compromised User
+path: /Basic Security Monitoring/Authentication Attacks
+description: This rule detects a successful login attempt from a compromised user account. 
+type: standard
+author: DNIF
+entity_type: workbooks
+blocks:
+- type: dql
+  query: stream=authentication where action='LOGIN' and status='PASSED' | groupby
```
Comment (aman1808) [inline workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml:10]: how do we know that the user is a compromised user, this does not seem to be a complete use case. please discuss this internally with @RoshanPoojari20

Comment (aman1808) [reply workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml:10]: should ideally have a pre existing list of compromised users from a event store to confirm this.

File: workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    user, srcip | limit 100
+- type: signal
+  DetectionName: Successfull Login from Compromised User
+  DetectionTactic: Initial Access
+  DetectionTechnique: Valid Accounts
+  DetectionScore: '5'
```
Comment (aman1808) [inline workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml:16]: the score and severity needs to be more accurate, after validation of compromised user

File: workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  DetectionSeverity: Medium
+  DetectionConfidence: Medium
+  SuspectHost: srcip
+  TargetUser: user
+  raw_event_query: SELECT * FROM AUTHENTICATION WHERE $Action="LOGIN" AND $Status="PASSED"
```
Comment (aman1808) [inline workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml:21]: use new dql here

**Review by aman1808 — CHANGES_REQUESTED** [review]

File: workbooks/Advanced Threat Detection/IAM Monitoring/mfa_disabled_for_gcp_user.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
+id: 06d3eef8a62e4d82b98fd1eea21918f5
+name: MFA Disabled for GCP User
+path: /Advanced Threat Detection/IAM Monitoring
+description: This rule detects when Multi-Factor Authentication (MFA) is disabled for a user in Google Cloud Platform (GCP). This modification weakens the security of the accounts and the attackers can keep persistance in the environment without adding new users.
+type: standard
+author: DNIF
+entity_type: workbooks
+blocks:
+- type: dql
+  query: stream=iam where action='UNENROLL_USER_FROM_STRONG_AUTH' | groupby user,
```
[the remark this answers is not in the document]
Comment (Avni3) [reply workbooks/Advanced Threat Detection/IAM Monitoring/mfa_disabled_for_gcp_user.yml]: Added the sourcename in the query as G-SUITE and increased the limit to 100

File: workbooks/Advanced Threat Detection/IAM Monitoring/mfa_disabled_for_gcp_user.yml (hunk 1 above)
Comment (Avni3) [reply workbooks/Advanced Threat Detection/IAM Monitoring/mfa_disabled_for_gcp_user.yml]: Updated the raw event query in new dql and added sourcename as G-SUITE

File: workbooks/Advanced Threat Detection/IAM Monitoring/mfa_disabled_for_gcp_user.yml (hunk 2 above)
Comment (Avni3) [reply workbooks/Advanced Threat Detection/IAM Monitoring/mfa_disabled_for_gcp_user.yml]: Changed the DetectionScore, DetectionSeverity, DetectionConfidence in the usecase

File: workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml (hunk 3 above)
Comment (Avni3) [reply workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml:10]: Discussed with Roshan and added a second query which is responsible for checking if the target user value is present in the Compromised User list. Also, created a retrieve compromised user usecase to maintain a list of target user which are found to be compromised.

File: workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml (hunk 4 above)
Comment (Avni3) [reply workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml]: Changed the DetectionScore, DetectionSeverity, and DetectionConfidence in the usecase.

File: workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml (hunk 5 above)
Comment (Avni3) [reply workbooks/Basic Security Monitoring/Authentication Attacks/successfull_login_from_compromised_user.yml]: Added New DQL in raw event query

**Review by aman1808 — APPROVED** [review]

**Pull request merged**